Sources/TokamakCore/Modifiers/AppearanceActionModifier.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+protocol AppearanceActionProtocol {
+  var appear: (() -> ())? { get }
+  var disappear: (() -> ())? { get }
+}
+
+/// Underscore is present in the name for SwiftUI compatibility.
+struct _AppearanceActionModifier: ViewModifier {
+  var appear: (() -> ())?
+  var disappear: (() -> ())?
+  init(appear: (() -> ())? = nil, disappear: (() -> ())? = nil) {
+    self.appear = appear
+    self.disappear = disappear
+  }
+
+  typealias Body = Never
+}
+
+extension ModifiedContent: AppearanceActionProtocol
+  where Content: View, Modifier == _AppearanceActionModifier {
+  var appear: (() -> ())? { modifier.appear }
+  var disappear: (() -> ())? { modifier.disappear }
+}
+
+extension View {
+  public func onAppear(perform action: (() -> Swift.Void)? = nil) -> some View {
```

Comment:
I'm guessing this came from the `swiftinterface` file, but `Swift.Void` can just be `Void`, I personally find that to be cleaner.

Reply:
Oh, wasn't intentional, gonna fix it now, great shout